File: themes/theme_manager.py
from PyQt5.QtWidgets import QWidget
from PyQt5.QtGui import QPalette, QColor

class ThemeManager:
    def __init__(self):
        self.themes = {
            "tokyo_night": get_tokyo_night_theme(),
            "aura": get_aura_theme(),
            "light": get_light_theme(),
            "anysphere": get_anysphere_theme()
        }

    def apply_theme(self, widget: QWidget, theme_name: str):
        if theme_name not in self.themes:
            return
        
        theme = self.themes[theme_name]
        widget.setStyleSheet(theme["main_widget"])
        return theme

def get_tokyo_night_theme():
    return {
        "main_widget": """
            QWidget {
                background-color: #1a1b26;
                color: #a9b1d6;
                border: none;
            }
            QTextEdit, QComboBox {
                background-color: #24283b;
                color: #a9b1d6;
                border: 1px solid #414868;
                border-radius: 5px;
            }
            QComboBox::drop-down {
                border: none;
            }
            QComboBox::down-arrow {
                image: url(path/to/down_arrow.png);
                width: 12px;
                height: 12px;
            }
            QPushButton {
                background-color: transparent;
                color: #ffffff;
                border: 1px solid #3d59a1;
                padding: 8px 15px;
                border-radius: 5px;
                font-weight: bold;
            }
            QPushButton:hover {
                background-color: rgba(61, 89, 161, 0.1);
                border: 1px solid #2ac3de;
            }
            QLabel {
                color: #a9b1d6;
            }
        """,
        "theme_button": """
            QPushButton {
                background-color: transparent;
                color: #ffffff;
                border: 1px solid #3d59a1;
                border-radius: 5px;
            }
            QPushButton:hover {
                background-color: rgba(42, 195, 222, 0.1);
                border: 1px solid #2ac3de;
            }
        """,
        "text_color": "#a9b1d6",
        "border_color": "#414868",
        "title_color": "#ffffff",
        "colors": {
            "background": "#1a1b26",
            "text": "#a9b1d6",
            "button": "#3d59a1",
            "button_text": "#ffffff",
            "button_hover": "#2ac3de",
            "input_bg": "#24283b",
            "border": "#414868",
            "title": "#ffffff"
        }
    }

def get_aura_theme():
    return {
        "main_widget": """
            QWidget {
                background-color: #15141b;
                color: #edecee;
                border: none;
            }
            QTextEdit, QComboBox {
                background-color: #15141b;
                color: #edecee;
                border: 1px solid #6d6d6d;
                border-radius: 5px;
            }
            QComboBox::drop-down {
                border: none;
            }
            QComboBox::down-arrow {
                image: url(path/to/down_arrow.png);
                width: 12px;
                height: 12px;
            }
            QPushButton {
                background-color: transparent;
                color: #edecee;
                border: 1px solid #a277ff;
                padding: 8px 15px;
                border-radius: 5px;
                font-weight: bold;
            }
            QPushButton:hover {
                background-color: rgba(162, 119, 255, 0.1);
                border: 1px solid #61ffca;
            }
            QLabel {
                color: #edecee;
            }
        """,
        "theme_button": """
            QPushButton {
                background-color: transparent;
                color: #edecee;
                border: 1px solid #a277ff;
                border-radius: 5px;
            }
            QPushButton:hover {
                background-color: rgba(97, 255, 202, 0.1);
                border: 1px solid #61ffca;
            }
        """,
        "text_color": "#edecee",
        "border_color": "#6d6d6d",
        "title_color": "#edecee",
        "colors": {
            "background": "#15141b",
            "text": "#edecee",
            "button": "#a277ff",
            "button_text": "#edecee",
            "button_hover": "#61ffca",
            "input_bg": "#15141b",
            "border": "#6d6d6d",
            "title": "#edecee"
        }
    }

def get_light_theme():
    return {
        "main_widget": """
            QWidget {
                background-color: #f0f0f0;
                color: #333333;
                border: none;
            }
            QTextEdit, QComboBox {
                background-color: #ffffff;
                color: #333333;
                border: 1px solid #cccccc;
                border-radius: 5px;
            }
            QComboBox::drop-down {
                border: none;
            }
            QComboBox::down-arrow {
                image: url(path/to/down_arrow_dark.png);
                width: 12px;
                height: 12px;
            }
            QPushButton {
                background-color: transparent;
                color: #4a90e2;
                border: 1px solid #4a90e2;
                padding: 8px 15px;
                border-radius: 5px;
                font-weight: bold;
            }
            QPushButton:hover {
                background-color: rgba(74, 144, 226, 0.1);
                border: 1px solid #357abd;
            }
            QLabel {
                color: #333333;
            }
        """,
        "theme_button": """
            QPushButton {
                background-color: transparent;
                color: #4a90e2;
                border: 1px solid #4a90e2;
                border-radius: 5px;
            }
            QPushButton:hover {
                background-color: rgba(53, 122, 189, 0.1);
                border: 1px solid #357abd;
            }
        """,
        "text_color": "#333333",
        "border_color": "#cccccc",
        "title_color": "#333333",
        "colors": {
            "background": "#f0f0f0",
            "text": "#333333",
            "button": "#4a90e2",
            "button_text": "#ffffff",
            "button_hover": "#357abd",
            "input_bg": "#ffffff",
            "border": "#cccccc",
            "title": "#333333"
        }
    }

def get_anysphere_theme():
    return {
        "main_widget": """
            QWidget {
                background-color: #0e1116;
                color: #e6edf3;
                border: none;
            }
            QTextEdit, QComboBox {
                background-color: #0e1116;
                color: #e6edf3;
                border: 1px solid #30363d;
                border-radius: 5px;
            }
            QComboBox::drop-down {
                border: none;
            }
            QComboBox::down-arrow {
                image: url(path/to/down_arrow.png);
                width: 12px;
                height: 12px;
            }
            QPushButton {
                background-color: transparent;
                color: #ffffff;
                border: 1px solid #E6C895;
                padding: 8px 15px;
                border-radius: 5px;
                font-weight: bold;
            }
            QPushButton:hover {
                background-color: rgba(230, 200, 149, 0.1);
                border: 1px solid #E6C895;
            }
            QLabel {
                color: #e6edf3;
            }
        """,
        "theme_button": """
            QPushButton {
                background-color: transparent;
                color: #ffffff;
                border: 1px solid #E6C895;
                border-radius: 5px;
            }
            QPushButton:hover {
                background-color: rgba(230, 200, 149, 0.1);
                border: 1px solid #E6C895;
            }
        """,
        "text_color": "#e6edf3",
        "border_color": "#30363d",
        "title_color": "#ffffff",
        "colors": {
            "background": "#0e1116",
            "text": "#e6edf3",
            "button": "#E6C895",
            "button_text": "#ffffff",
            "button_hover": "#E6C895",
            "input_bg": "#0e1116",
            "border": "#30363d",
            "title": "#ffffff"
        }
    }

File: VorTeX_frontend/VorTeX_frontend/ThemeManager.swift
import SwiftUI

// Theme structure
struct ThemeColors {
    let background: Color
    let text: Color
    let button: Color
    let buttonText: Color
    let buttonHover: Color
    let inputBackground: Color
    let border: Color
    let title: Color
}

class ThemeManager {
    static let shared = ThemeManager()
    
    // Theme dictionary
    private let themes: [String: ThemeColors] = [
        "tokyo_night": ThemeColors(
            background: Color(hex: "#1a1b26"),
            text: Color(hex: "#a9b1d6"),
            button: Color(hex: "#3d59a1"),
            buttonText: Color(hex: "#ffffff"),
            buttonHover: Color(hex: "#2ac3de"),
            inputBackground: Color(hex: "#24283b"),
            border: Color(hex: "#414868"),
            title: Color(hex: "#ffffff")
        ),
        "aura": ThemeColors(
            background: Color(hex: "#15141b"),
            text: Color(hex: "#edecee"),
            button: Color(hex: "#a277ff"),
            buttonText: Color(hex: "#edecee"),
            buttonHover: Color(hex: "#61ffca"),
            inputBackground: Color(hex: "#15141b"),
            border: Color(hex: "#6d6d6d"),
            title: Color(hex: "#edecee")
        ),
        "light": ThemeColors(
            background: Color(hex: "#f0f0f0"),
            text: Color(hex: "#333333"),
            button: Color(hex: "#4a90e2"),
            buttonText: Color(hex: "#ffffff"),
            buttonHover: Color(hex: "#357abd"),
            inputBackground: Color(hex: "#ffffff"),
            border: Color(hex: "#cccccc"),
            title: Color(hex: "#333333")
        ),
        "anysphere": ThemeColors(
            background: Color(hex: "#0e1116"),
            text: Color(hex: "#e6edf3"),
            button: Color(hex: "#E6C895"),
            buttonText: Color(hex: "#ffffff"),
            buttonHover: Color(hex: "#E6C895"),
            inputBackground: Color(hex: "#0e1116"),
            border: Color(hex: "#30363d"),
            title: Color(hex: "#ffffff")
        ),
        "system": ThemeColors(
            background: Color(NSColor.windowBackgroundColor),
            text: Color(NSColor.textColor),
            button: Color(NSColor.controlAccentColor),
            buttonText: Color(NSColor.textBackgroundColor),
            buttonHover: Color(NSColor.controlAccentColor).opacity(0.8),
            inputBackground: Color(NSColor.textBackgroundColor),
            border: Color(NSColor.separatorColor),
            title: Color(NSColor.labelColor)
        )
    ]
    
    func getThemeColors(for themeKey: String) -> ThemeColors {
        return themes[themeKey] ?? themes["anysphere"]!
    }
}

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (1, 1, 1, 0)
        }

        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
} 